sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -180,10 +180,30 @@ class JsonFunctionsSuite extends QueryTest with SharedSQLContext {
 
   test("to_json - array") {
     val df = Seq(Tuple1(Tuple1(1) :: Nil)).toDF("a")
+    val df2 = Seq(Tuple1(Map("a" -> 1) :: Nil)).toDF("a")
 
     checkAnswer(
       df.select(to_json($"a")),
       Row("""[{"_1":1}]""") :: Nil)
+    checkAnswer(
+      df2.select(to_json($"a")),
+      Row("""[{"a":1}]""") :: Nil)
+  }
+
+  test("to_json - map") {
+    val df1 = Seq(Map("a" -> Tuple1(1))).toDF("a")
+    val df2 = Seq(Map(Tuple1(1) -> Tuple1(1))).toDF("a")
+    val df3 = Seq(Map("a" -> 1)).toDF("a")
+
+    checkAnswer(
+      df1.select(to_json($"a")),
+      Row("""{"a":{"_1":1}}""") :: Nil)
+    checkAnswer(
+      df2.select(to_json($"a")),
+      Row("""{"[0,1]":{"_1":1}}""") :: Nil)
```

Comment:
This case looks rather a string representation from `UnsafeRow` and a bug. Let's remove this test case for now and fix it later together in another PR.

Reply:
I think we don't have a `toString` method for `UnsafeRow` to show up well-format string representation. The current `toString` for it is just for debugging. It simply gets the hex string for each 8 bytes. Consider the nature of `UnsafeRow`, this `toString` might make sense to me.

We can remove this test case if the result could be confusing, otherwise we should add a comment to explain why the JSON looks like it.

Reply:
Yes, I was thinking that showing debugging (internal formatted) strings is rather a bug and actually we might have to fix if possible, as I think we have pretty `toString` format for public API `Row` class.

I remember we eventually decided to fix `TimestampType` and `DataType` in JSON, which were printed as longs and integers, to human readable formats, SPARK-16216. I think there was a discussion that they should be written as are (long and int) vs human readable format at that time if I understood correctly.

Reply:
I get your point and I know that also makes sense. How about removing this case out away? I think we might have to discuss more if this is a bug or not but want to rather take out what we are not sure of for now (as it is strictly not related with this PR, if I understood correctly).

Reply:
Yeah, I agree to remove it for now.